pkg/k8s/watchers/pod.go

```diff
@@ -719,6 +719,11 @@ func (k *K8sWatcher) deleteHostPortMapping(pod *slim_corev1.Pod, podIPs []string
 	}
 
 	for _, dpSvc := range svcs {
+		svcs := k.svcManager.GetDeepCopyServices()
```

Comment:
Copying all of these, in each loop iteration, seems like it might be a bit heavy?

Reply:
Good catch! Yes, there is definitely no need for the service copies to be retrieved in each iteration, and the operation can definitely be moved outside the for-loop. Will go ahead and do so 👍

Reply:
commented below; we can simplify this even more :-)

Reply:
done